builtin/logical/mysql/secret_creds.go

```diff
@@ -59,27 +81,44 @@ func (b *backend) secretCredsRevoke(
 		return nil, err
 	}
 
+	// Get the role name
+	roleNameRaw, ok := req.Secret.InternalData["role"]
```

Comment:
You're not checking `ok` here so if the value isn't there (upgrade case) it'll be `nil`. This will cause a panic on the next line where you type-assert to a `string`. You should declare `roleName` ahead of time (`var roleName string`), then the next line should only happen if `ok` is true (and should be `=` not `:=` in that case).